### PR TITLE
Gci 1303 send mid to chd order api

### PR DESCRIPTION
**Pull request by nabeelamirch**
send mid to chd order api

resolves: GCI-1303

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
 package uk.gov.companieshouse.chdorderconsumer.service;
 
+import org.springframework.http.HttpStatus;
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java:3]: See http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Achd-order-consumer&pullRequest=10&resolved=false&types=CODE_SMELL.

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 import uk.gov.companieshouse.api.InternalApiClient;
 import uk.gov.companieshouse.api.error.ApiErrorResponseException;
 import uk.gov.companieshouse.api.handler.exception.URIValidationException;
 import uk.gov.companieshouse.api.model.ApiResponse;
 import uk.gov.companieshouse.api.model.order.chd.MissingImageDeliveryRequestApi;
+import uk.gov.companieshouse.chdorderconsumer.exception.RetryableErrorException;
 import uk.gov.companieshouse.chdorderconsumer.exception.ServiceException;
 
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
```
Comment (caismith-ch) [inline src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java:14]: Few unused imports in this file that can be removed

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        } catch (ApiErrorResponseException ex) {
+            String id = missingImageDeliveryRequestApi.getId() != null ? missingImageDeliveryRequestApi.getId() : "Unknown";
+            String errorResponse = "API Response Error for : "
+                    + id + ", Error response: " + ex.toString();
+            processException(ex.getStatusCode(), errorResponse);
+        }
+    }
+
+    public void processException(int statusCode, String errorResponse) {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java:68]: Perhaps `processError()`? Does it need to be public?

File: start.sh [hunk 4]
```diff
@@ -27,5 +27,6 @@ else
     source "${APP_DIR}/app_env"
 fi
 
-exec java ${JAVA_MEM_ARGS} -jar -Dserver.port="${PORT}" "${APP_DIR}/chd-order-consumer.jar"
+#exec java ${JAVA_MEM_ARGS} -jar -Dserver.port="${PORT}" "${APP_DIR}/chd-order-consumer.jar"
+exec mvn spring-boot:run -Dspring-boot.run.jvmArguments="-Dserver.port=18577"
```
Comment (caismith-ch) [inline start.sh]: Revert to original start up line and remove debug line

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java (hunk 1 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java]: Ty fixed all the smelly code

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java (hunk 2 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/CHDOrderService.java]: Ty fixed all the smelly code

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java (hunk 3 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java:67]: Yes, I'll change that now, It's not actually processing an exception, it throws it.

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    private static final String POST_MISSING_IMAGE_CHD_ORDER_URI = "/chd-order-api/missing-image-deliveries";
+
+    private final CHDOrderService chdOrderService;
+
+    public ItemOrderedProcessorService(final CHDOrderService chdOrderService) {
+        this.chdOrderService = chdOrderService;
+    }
+
+    public MissingImageDeliveryRequestApi mapChdItemOrderedToMissingImageDeliveryRequestApi(ChdItemOrdered chdItemOrdered) {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java:30]: Does this need to be public?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java:30]: Yes, but for the unit test.

File: src/test/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        assertThat(missingImageDeliveryRequestApi.getFilingHistoryType(), is(CHD_ITEM_ORDERED.getItem().getItemOptions().get("filingHistoryType")));
+        assertThat(missingImageDeliveryRequestApi.getItemCost(), is(CHD_ITEM_ORDERED.getItem().getTotalItemCost()));
+    }
+
+    @Test
+    void propogatesRetryableExceptionIfApiErrorResponseExceptionIsInternalServerError() throws ApiErrorResponseException {
+        when(chdOrderService.createCHDOrder(anyString(), any(MissingImageDeliveryRequestApi.class)))
+                .thenThrow(buildApiErrorResponseException(INTERNAL_SERVER_ERROR));
+
+        assertThrows(RetryableErrorException.class, () ->
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorServiceTest.java:85]: Could these be extended to assert the expected exception message too?

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java (hunk 5 above)
Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java:30]: Could use package (default) access level for that.

**Review by VaughanJackson123 — APPROVED** [review]

**Review by caismith-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java (hunk 5 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java]: I have changed this

**Pull request merged**